contrib/move-string-resource.kts

```diff
@@ -0,0 +1,167 @@
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/**
+ * USAGE:
+ * Cuts string resources with the name arg3 from the module from arg1
+ * and pastes them to the string resources from the module from arg2.
+ *
+ * Example usage:
+ * kotlinc -script move-string-resource.kts ../app ../core history_from_current_book
+ *
+ * This will cut all strings with the name history_from_current_book from
+ * all strings.xml files within directories with a prefix of values*
+ * in ../app/src/main/res/ and paste them into
+ * their corresponding files within ../core/src/main/res/values*.
+ *
+ */
+import java.io.File
+import kotlin.system.exitProcess
+
+
+if (args.size < 3) {
+  printCorrectUsageAndExit()
+}
+
+val pathToValues = "/src/main/res/"
+val source = args[0]
+val destination = args[1]
+
+/// name=key
+val key = """name="${args[2]}""""
+
+val sourceDir = File(source + pathToValues)
+if (!sourceDir.exists()) {
+  printModuleIsNotValidDir(source, pathToValues)
+}
+
+println("Running transfer of string resources...")
+val numberOfFilesPastedTo = sourceDir.cutStringResourcesAndPasteToDestination(key, source, destination)
+println("Transfer of string resources complete. Moved $numberOfFilesPastedTo strings.")
+
+fun File.cutStringResourcesAndPasteToDestination(key: String, source: String, destination: String) : Int {
+  var numberOfFilesPastedTo = 0
+  this.walk().filter { it.name.equals("strings.xml") }.forEach { resourceFile ->
+    var cutLine: String = cutLineFromResourceFile(resourceFile, key)
+    if (cutLine.equals("")) {
+      // continue for each loop
+      return@forEach
+    }
+    val destinationFile = openOrCreateDestinationResourceFile(resourceFile, source, destination)
+    pasteLineToDestination(destinationFile, cutLine)
+    numberOfFilesPastedTo++
+  }
+  return numberOfFilesPastedTo
+}
+
+fun openOrCreateDestinationResourceFile(resourceFile: File, source: String, destination: String): File {
+
+  val destinationFilePath = resourceFile.path.replaceRange(0..source.length, destination + "/")
+  val destinationFile = File(destinationFilePath)
+  val destinationDirectoryPath = destinationFilePath
+    .removeRange(
+      (destinationFilePath.length - "strings.xml".length - 1)..destinationFilePath.length - 1
+    )
+  val destinationDirectory = File(destinationDirectoryPath)
+
+  if (!destinationDirectory.exists()) {
+    createDestinationDirectoryAndFile(destinationDirectory, destinationFile)
+  }
+  return destinationFile
+}
+
+fun createDestinationDirectoryAndFile(destinationDirectory: File, destinationFile: File) {
+  destinationDirectory.mkdir()
+  val isNewFileCreated: Boolean = destinationFile.createNewFile()
+  if (!isNewFileCreated) {
+    System.err.println("Could not create resource file ${destinationFile.path}")
+    System.exit(-1)
+  }
+  val initialXMLResourceFileContent = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n" +
```

Comment:
here we can definitely use a raw string essentailly what they are made for.
"""
<?xml version=\"1.0\" encoding=\"UTF-8\"?>
<resources>
</resources>
""".trimIdent()

Reply:
I hadn't formatted this properly so it was hiding the contents of the raw string, edited